src/ColorTypes.jl

```diff
@@ -31,7 +32,7 @@ export RGB24, ARGB32, Gray24, AGray32
 
 
 ## Functions
-export base_color_type, base_colorant_type, ccolor, color, color_type
+export base_color_type, base_colorant_type, floattype, ccolor, color, color_type
```

Comment:
FixedPointNumbers doesn't export `floattype`, this should be consistent.

Reply:
I plan to remove the direct dependent on `ColorTypes.jl` or `FixedPointNumbers.jl` for `Images.jl`.

Perhaps without `import FixedPointNumbers: floattype`, just define a new function with the same name and export it? I.e.,

```julia
export floattype

floattype(::Type{T}) where T<:AbstractFloat = T
floattype(::Type{T}) where T<:FixedPoint = FixedPointNumbers.floattype(T)
```

If FixedPointNumbers doesn't intend to export this name, it would be good though.

Reply:
> I plan to remove the direct dependent on ColorTypes.jl or FixedPointNumbers.jl for Images.jl

For example by re-exporting the types from ImageCore? There might be some merit in that; if so we could export `floattype` from FixedPointNumbers.

Reply:
Yes, since ImageCore --reexport--> Colors --Reexport--> ColorTypes.

I can open a PR in FixedPointNumbers if you think this is worth doing.

Reply:
An alternative, `floattype_colorant` can be

```julia
_floattype(::Type{T}) where T <: AbstractFloat = T
_floattype(::Type{T}) where T <: Bool = Float32
_floattype(::Type{T}) where T <: FixedPoint = FixedPointNumbers.floattype(T)
floattype_colorant(::Type{T}) where T <: Colorant = base_colorant_type(T){_floattype(T)}
```

Reply:
The whole purpose of `floattype` is to return a value that safe to act as an accumulator without overflow; that's what this function is doing, so it should be called `floattype`.